### PR TITLE
add net.puddletag.puddletag

### PR DESCRIPTION
**Pull request by raduzaharia-medium**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:https://github.com/puddletag/puddletag/issues/601#**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: net.puddletag.puddletag.yaml
Comment (hfiguiere) [inline net.puddletag.puddletag.yaml:19]:
```suggestion
```

File: net.puddletag.puddletag.yaml
Comment (hfiguiere) [inline net.puddletag.puddletag.yaml:27]:
```suggestion
```

This probably trigger a warning with flatpak builder anyway.

**Comment by JakobDev** [comment]
You need a Appstream file

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
bot, build net.puddletag.puddletag

File: net.puddletag.puddletag.yaml
Comment (raduzaharia-medium) [reply net.puddletag.puddletag.yaml]: Ok, updated the yaml.

File: net.puddletag.puddletag.yaml
Comment (raduzaharia-medium) [reply net.puddletag.puddletag.yaml]: Right, it was a leftover - fixed.

File: net.puddletag.puddletag.yaml
Comment (raduzaharia-medium) [reply net.puddletag.puddletag.yaml]: Fixed.

**Comment by raduzaharia-medium** [comment]
> You need a Appstream file

Ok, added the appdata file too.

File: net.puddletag.puddletag.appdata.xml
Comment (hfiguiere) [inline net.puddletag.puddletag.appdata.xml:48]: `content_rating` is missing

the file is also never referenced nor installed in the manifest.

Comment (raduzaharia-medium) [reply net.puddletag.puddletag.appdata.xml:48]: Ok, added both.

File: net.puddletag.puddletag.yaml
Comment (hfiguiere) [inline net.puddletag.puddletag.yaml:80]: it's never installed. That's why the build fails

Comment (raduzaharia-medium) [reply net.puddletag.puddletag.yaml:81]: Added the metainfo installation step.

**Comment by hfiguiere** [comment]
bot, build net.puddletag.puddletag

File: net.puddletag.puddletag.appdata.xml
Comment (hfiguiere) [inline net.puddletag.puddletag.appdata.xml:17]: a bare amperand in XML is always bad. This is what causes the build failure.

Also I'm not even sure this is a valid category.
Valid categories are https://specifications.freedesktop.org/menu-spec/latest/apa.html

Comment (raduzaharia-medium) [reply net.puddletag.puddletag.appdata.xml]: Yup, fixed.

**Comment by hfiguiere** [comment]
bot, build net.puddletag.puddletag

**Comment by tinywrkb** [comment]
It shouldn't be merged as it is without checking every usage of QFileDialog which clearly wasn't done. At least, the playlist loading and saving dialogs need to set `DontUseNativeDialog` as true.  
As long as we don't have the "open neighboring files" portal, then this workaround is needed.

**Comment by raduzaharia-medium** [comment]
> It shouldn't be merged as it is without checking every usage of QFileDialog which clearly wasn't done. At least, the playlist loading and saving dialogs need to set `DontUseNativeDialog` as true. As long as we don't have the "open neighboring files" portal, then this workaround is needed.

Ooouch...ok. Let me ask the developers if they agree with the change. If not I guess I could make a fork of the project but it will get messy. I'll be right back with an answer, please don't close the pull request.

**Comment by raduzaharia-medium** [comment]
Ready for another test run please :)

**Comment by raduzaharia-medium** [comment]
bot, build net.puddletag.puddletag

**Comment by raduzaharia-medium** [comment]
@tinywrkb @nedrichards @hfiguiere 
I tested the app, seems to work as expected. By adding some flatseal permissions I was even able to access the NFS shares and edit ID3 tags straight from there. For all I can say, it works just like the regular app.

**Comment by tinywrkb** [comment]
As a fellow maintainer, I don't expect this to be approved with downstream patches.  
As one who maintained a couple of orphaned apps whose original submitters are MIAs, the last thing I want to see in an application is a downstream patch.  

I suggest that you communicated with the application's developers on how to upstream the needed adjustments for Flatpak.  
A possible way to detect the Flatpak environment is to check for the existence of `/.flatpak-info`.  
If you need an example on how to handle this in a PyQt app, then look what we have in [qutebrowser](https://github.com/qutebrowser/qutebrowser/), which also a GPL3 license app. Search for `_FLATPAK_INFO_PATH`, and `is_flatpak`.

**Comment by raduzaharia-medium** [comment]
@tinywrkb I hear you but in this case I agreed with the app developer to proceed this way. The patch will probably no longer be needed anyway once flatpak develops more. I also found many app submissions to flathub applying patches, many exactly for this situation. Please allow the submission. If the app will fade due to developers gone MIA, we can remove it from flatpak then.

**Comment by tinywrkb** [comment]
@raduzaharia-medium note that this is just my personal opinion as a fellow maintainer. You'll have to wait for a reply with Flathub associated developer.

File: net.puddletag.puddletag.appdata.xml
Comment (barthalion) [inline net.puddletag.puddletag.appdata.xml:1]: This file should be submitted and maintained upstream.

File: dontusenativedialog.patch
Comment (barthalion) [inline dontusenativedialog.patch:1]: Has this patch been submitted upstream?

File: net.puddletag.puddletag.yaml
Comment (barthalion) [inline net.puddletag.puddletag.yaml:86]:
```suggestion
        buildsystem: cmake-ninja
```

File: dontusenativedialog.patch
Comment (raduzaharia-medium) [reply dontusenativedialog.patch:1]: Yes, but they won't allow it. The developer don't want to maintain any kind of flatpak runtime changes. That's the only reason I am using an external patch that I am applying on the flatpak build.

File: net.puddletag.puddletag.appdata.xml
Comment (raduzaharia-medium) [reply net.puddletag.puddletag.appdata.xml:1]: As for the patch, the developer does not want to have the file, that's the only reason I included it here.

**Comment by raduzaharia-medium** [comment]
bot, build net.puddletag.puddletag

File: dontusenativedialog.patch
Comment (barthalion) [reply dontusenativedialog.patch:1]: Do you have an URL to said discussion?

Comment (barthalion) [reply dontusenativedialog.patch:1]: OK, it's at https://github.com/puddletag/puddletag/issues/735. I would recommend following up on this with @tinywrkb suggestion to make this dynamic based on `/.flatpak-info` presence.

Comment (raduzaharia-medium) [reply dontusenativedialog.patch:1]: yes: https://github.com/puddletag/puddletag/issues/735

**Pull request closed without merging**
